Update storage format docs (#2309)
grafana/pyroscope: docs/sources/reference-pyroscope-architecture/block-format/_index.md

* A [TSDB index] `index.tsdb` mapping the external labels to the profiles
  stored in the profiles table.

* `profiles.parquet` [parquet] table that contains profiles.

* `symbols` sub-directory contains profiling symbols that provide a link between 
  the compiled or interpreted binary code and the original source code:
   - A `index.symdb` file with meta information, which helps to find symbols for a specific profile.
   - A `stacktraces.symdb` file contains stack traces compacted in the [parent pointer tree].
   - Parquet tables for models referenced by stack traces:
     `locations.parquet`, `functions.parquet`, `mappings.parquet`, `strings.parquet`.

## Data model

The data model within the block is fairly aligned to Google's [proto
definition][pprof] for the pprof wire format.

> Profile labels contain additional information gathered at ingestion time and can

Reviewer: Maybe that's worth it?
```suggestion
Profile series labels contain additional information gathered at ingestion time and can
```

> be used to select certain profiles. They are comparable to Prometeus'/Loki's label

Reviewer: ```suggestion
be used to select certain profiles. They are comparable to Prometeus'/Loki's labels
```

> and typical labels are `namespace` and `pod` to describe which workload the profiles

Reviewer: ```suggestion
and typical label names are `namespace` and `pod` to describe which workload the profiles
```

> are coming form.

Author: We no longer store pprof labels

> are coming form.

Reviewer: ```suggestion
are coming from.
```
